File: stock-ms/src/main/java/br/puc/tp_final/stock/StockController.java
package br.puc.tp_final.stock;

import jakarta.ejb.EJB;
import jakarta.ws.rs.*;

@Path("stock")
@Produces({"application/json"})
public class StockController {

    @EJB
    private StockService stockService;

    @POST
    @Path("write_off")
    @Consumes("application/json")
    public void pay() {
        stockService.write_off();
    }

    @GET
    @Path("/status/{id}")
    @Consumes("application/json")
    public String status() {
        return stockService.status();
    }
}

File: stock-ms/src/main/kotlin/br/puc/tp_final/stock/StockController.kt
package br.puc.tp_final.stock

import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/stock-ms/rest/stock")
class StockController(
    val stockService: StockService
) {
    @PostMapping("write_off")
    fun writeOff() {
        stockService.writeOff()
    }

    @GetMapping("/status/{id}")
    fun status(): String {
        return stockService.status();
    }
}